### PR TITLE
Add a Makefile to replace build.sh and test.sh.

### PR DESCRIPTION
**Pull request by dlorenc**
This creates some make actions to build and test:

make minikube
make localkube
make minikube-linux-amd64, etc.
make localkube-linux-amd64, etc.

make test
make integration

### CODE REVIEW

**Comment by dlorenc** [comment]
cc @ethernetdan Could you take a look here when you get a chance?

**Comment by dlorenc** [comment]
cc @luxas 
ref #43

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,43 @@
+GO ?= go
+GOOS ?= $(shell go env GOOS)
+GOARCH ?= $(shell go env GOARCH)
+BUILD_DIR ?= ./out
+GOPATH := $(shell pwd)/.gopath
+REPOPATH ?= k8s.io/minikube
+
```
Comment (luxas) [inline Makefile]: GO15VENDOREXPERIMENT=1

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+.PHONY: clean
```
Comment (luxas) [inline Makefile]: Why all these PHONYs?

File: Makefile (hunk 1 above)
Comment (dlorenc) [inline Makefile]: Thanks! I've been using Go 1.6 for everything so I forget to set that.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+localkube: localkube-$(GOOS)-$(GOARCH)
+	cp $(BUILD_DIR)/localkube-$(GOOS)-$(GOARCH) $(BUILD_DIR)/localkube
+
+.PHONY: minikube-$(GOOS)-$(GOARCH)
+minikube-$(GOOS)-$(GOARCH): gopath
+	GOPATH=$(GOPATH) CGO_ENABLED=0 GOARCH=$(GOARCH) GOOS=$(GOOS) $(GO) build --installsuffix cgo -a -o $(BUILD_DIR)/minikube-$(GOOS)-$(GOARCH) ./cmd/minikube
+
+.PHONY: localkube-$(GOOS)-$(GOARCH)
+localkube-$(GOOS)-$(GOARCH): gopath
+	GOARCH=$(GOARCH) GOOS=$(GOOS) $(GO) build -o $(BUILD_DIR)/localkube-$(GOOS)-$(GOARCH) -i --ldflags '-extldflags "-static" --s -w' ./cmd/localkube
```
Comment (luxas) [inline Makefile]: I think `localkube` should be dynamically built, since we want and need cAdvisor.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,44 @@
+GO ?= go
```
Comment (luxas) [inline Makefile]: Why this var?

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+.PHONY: localkube-$(GOOS)-$(GOARCH)
+localkube-$(GOOS)-$(GOARCH): gopath
+	GOARCH=$(GOARCH) GOOS=$(GOOS) $(GO) build -o $(BUILD_DIR)/localkube-$(GOOS)-$(GOARCH) -i --ldflags '-extldflags "-static" --s -w' ./cmd/localkube
+
+integration: minikube
+	$(GO) test -v ./integration --tags=integration
+
+.PHONY: test
+test: gopath
```
Comment (luxas) [inline Makefile]: What if the test code increases in complexity and gets larger?

**Comment by dlorenc** [comment]
PTAL @luxas, cleaned it up a bit. Thanks for the detailed review!

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+.PHONY: localkube
+localkube: localkube-$(GOOS)-$(GOARCH)
+	cp $(BUILD_DIR)/localkube-$(GOOS)-$(GOARCH) $(BUILD_DIR)/localkube
+
+minikube-$(GOOS)-$(GOARCH): gopath
+	GOPATH=$(GOPATH) CGO_ENABLED=0 GOARCH=$(GOARCH) GOOS=$(GOOS) go build --installsuffix cgo -a -o $(BUILD_DIR)/minikube-$(GOOS)-$(GOARCH) ./cmd/minikube
+
+localkube-$(GOOS)-$(GOARCH): gopath
+	GOARCH=$(GOARCH) GOOS=$(GOOS) CGO_ENABLED=1 go build -o $(BUILD_DIR)/localkube-$(GOOS)-$(GOARCH) -i ./cmd/localkube
```
Comment (ethernetdan) [inline Makefile]: Do we want to support platforms other than Linux for localkube?

Comment (dlorenc) [inline Makefile]: Probably not, but it is really nice to be able to run "make localkube" on my mac just to be sure everything compiles :)

**Pull request merged**